magento/magento2#28568:GraphQL query returns admin option value label…

gallyamov opened the pull request:
### Description (*)

This is for #28568 and #28572 

### Related Pull Requests

N/A

### Fixed Issues (if relevant)

1. Fixes magento/magento2#28568: Query returns admin option value label within aggregations
2. Fixes magento/magento2#28572: Custom attribute aggregations should return store specific option values

### Manual testing scenarios (*)

Check fixed issue for testing steps and expected results.

### Questions or comments

N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

gallyamov committed:
magento/magento2#28568:GraphQL query returns admin option value label within aggregations

- Added filter by store_id.

danielrenaud commented on app/code/Magento/CatalogGraphQl/DataProvider/Product/LayeredNavigation/AttributeOptionProvider.php:
```diff
@@ -42,15 +42,17 @@ public function __construct(ResourceConnection $resourceConnection)
             return [];
         }
 
+        $storeId = $storeId ?: 0;
```
Let's change "0" to use \Magento\Store\Model\Store::DEFAULT_STORE_ID instead

danielrenaud commented on app/code/Magento/CatalogGraphQl/DataProvider/Product/LayeredNavigation/AttributeOptionProvider.php:
```diff
@@ -42,15 +42,17 @@ public function __construct(ResourceConnection $resourceConnection)
      *
      * @param array $optionIds
      * @param array $attributeCodes
+     * @param int|null $storeId
```
Please swap params so the match argument order

danielrenaud reviewed (changes requested):
@gallyamov  Can you please add api-functional test cases to cover this change

lbajsarowicz commented on app/code/Magento/CatalogGraphQl/DataProvider/Product/LayeredNavigation/AttributeOptionProvider.php:
```diff
@@ -42,15 +42,17 @@ public function __construct(ResourceConnection $resourceConnection)
      * @return array
      * @throws \Zend_Db_Statement_Exception
      */
-    public function getOptions(array $optionIds, array $attributeCodes = []): array
```
What is the reason of adding parameter (optional) before other optional parameter

gallyamov replied:
Correct me if I am wrong, but in our case Store ID is not optional but required. Optional is only $attributeCodes. This is why I added Store ID before it.

lbajsarowicz replied:
The question mark before type means that you allow null there

gallyamov replied:
Yes, but as I know it is not optional, we have to pass int or null (correct me if I am wrong).

I am passing it with a question mark because the method I am calling it from also has it in params with a question mark. 

```
/**
     * Get list of attributes with options
     *
     * @param AggregationInterface $aggregation
     * @param int|null $storeId
     * @return array
     * @throws \Zend_Db_Statement_Exception
     */
    private function getAttributeOptions(AggregationInterface $aggregation, ?int $storeId): array
    {
        $attributeOptionIds = [];
        $attributes = [];
        foreach ($this->getAttributeBuckets($aggregation) as $bucket) {
            $attributes[] = \preg_replace('~_bucket$~', '', $bucket->getName());
            $attributeOptionIds[] = \array_map(
                function (AggregationValueInterface $value) {
                    return $value->getValue();
                },
                $bucket->getValues()
            );
        }

        if (!$attributeOptionIds) {
            return [];
        }

        return $this->attributeOptionProvider->getOptions(\array_merge(...$attributeOptionIds), $storeId, $attributes);
    }
```

lbajsarowicz replied:
Now I understand. Thank you for your effort to explain that!

gallyamov replied:
No problem, thanks!

gallyamov replied:
@lbajsarowicz could you mark this as resolved, please? It is blocking us to merge this fix.

lbajsarowicz reviewed (changes requested):
Store ID is optional, so you should add this after already defined parameters

gallyamov committed:
Merge branch '2.4-develop' of github.com:gallyamov/magento2 into 28568_admin_option_labels

gallyamov committed:
magento/magento2#28568:GraphQL query returns admin option value label within aggregations
- Code review fixes
-Added test case for store specific attribute option labels

gallyamov committed:
Merge branch '2.4-develop' of github.com:magento/magento2 into 28568_admin_option_labels

gallyamov commented:
@magento run all tests

gallyamov commented:
@magento run Functional Tests B2B

gallyamov commented:
@magento run Functional Tests B2B

Commit:
Merge branch '2.4-develop' into 28568_admin_option_labels

danielrenaud commented:
Review and QA passed

lbajsarowicz reviewed (approved):
:heavy_check_mark:

magento-engcom-team commented:
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7677](https://jira.corp.magento.com/browse/ENGCOM-7677) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

magento-engcom-team commented:
@gallyamov thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

gallyamov commented:
@magento run all tests

Commit:
Merge branch '2.4-develop' into 28568_admin_option_labels

gallyamov commented:
@magento run all tests

gallyamov commented:
@magento run Functional Tests B2B

gallyamov commented:
@magento run Functional Tests EE
